Kuroba/app/src/main/java/com/github/adamantcheese/chan/features/gesture_editor/Android10GesturesExclusionZonesHolder.kt

```diff
@@ -0,0 +1,236 @@
+
+        // The old settings must belong to a phone with the same screen size as the current one,
+        // otherwise something may break
+        if (
+                exclusionZones.minScreenSize != minScreenSize
+                || exclusionZones.maxScreenSize != maxScreenSize
+        ) {
+            val sizesString = "oldMinScreenSize = ${exclusionZones.minScreenSize}, " +
+                    "currentMinScreenSize = ${minScreenSize}, " +
+                    "oldMaxScreenSize = ${exclusionZones.maxScreenSize}, " +
+                    "currentMaxScreenSize = ${maxScreenSize}"
+
+            Logger.d(TAG, "Screen sizes do not match! $sizesString")
+            resetZones()
+            return zones
+        }
+
+        if (exclusionZones.zones.isEmpty()) {
+            return zones
+        }
+
+        exclusionZones.zones.forEach { zoneJson ->
+            if (!zones.containsKey(zoneJson.screenOrientation)) {
+                zones[zoneJson.screenOrientation] = mutableSetOf()
+            }
+
+            val zoneRect = ExclusionZone(
+                    screenOrientation = zoneJson.screenOrientation,
+                    attachSide = AttachSide.fromInt(zoneJson.attachSide),
+                    left = zoneJson.left.toInt(),
+                    right = zoneJson.right.toInt(),
+                    top = zoneJson.top.toInt(),
+                    bottom = zoneJson.bottom.toInt()
+            )
+
+            zoneRect.checkValid()
+            zones[zoneJson.screenOrientation]!!.add(zoneRect)
+            Logger.d(TAG, "Loaded zone ${zoneJson}")
+        }
+
+        return zones
+    }
+
+    fun addZone(orientation: Int, attachSide: AttachSide, zoneRect: Rect) {
+        if (!exclusionZones.containsKey(orientation)) {
+            exclusionZones[orientation] = mutableSetOf()
+        }
+
+        val exclusionZone = ExclusionZone(
+                screenOrientation = orientation,
+                attachSide = attachSide,
+                left = zoneRect.left,
+                right = zoneRect.right,
+                top = zoneRect.top,
+                bottom = zoneRect.bottom
+        )
+        exclusionZone.checkValid()
+
+        val prevZone = getZoneOrNull(orientation, attachSide)
+        if (prevZone != null) {
+            Logger.d(TAG, "addZone() Removing previous zone with the same params " +
+                    "as the new one, prevZone = ${prevZone}")
+            removeZone(prevZone.screenOrientation, prevZone.attachSide)
+        }
+
+        if (exclusionZones[orientation]!!.add(exclusionZone)) {
+            val newExclusionZones = mutableListOf<ExclusionZoneJson>()
+
+            exclusionZones.forEach { (orientation, zones) ->
+                zones.forEach { zone ->
+                    newExclusionZones += ExclusionZoneJson(
+                            screenOrientation = orientation,
+                            attachSide = zone.attachSide.id,
+                            left = zone.left.toFloat(),
```

Comment:
You're saving these as Floats, but up above you're reading them as Ints. Did you mean to do that?

Reply:
Done.